src/NUnitFramework/framework/ExceptionExtensions.cs

```diff
@@ -0,0 +1,102 @@
+        // race condition between threads reading the same PDB file to obtain file and line info for a stack trace
+        // results in AccessViolationException when the stack trace is accessed even indirectly e.g. Exception.ToString.
+        [System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptions]
+#endif
+        public static string GetStackTraceWithoutThrowing(this Exception exception)
+        {
+            if (exception is null) throw new ArgumentNullException(nameof(exception));
+
+            try
+            {
+                return exception.StackTrace;
+            }
+            catch (Exception ex)
+            {
+                return ex.GetType().Name + " was thrown by the Exception.StackTrace property.";
+            }
+        }
+
+        /// <summary>
+        /// If <see cref="Exception.Message"/> throws, returns "SomeException was thrown by the Exception.Message
+        /// property."
+        /// </summary>
+#if !(NET35 || NETSTANDARD1_4)
+        // https://github.com/dotnet/coreclr/issues/19698 is also currently present in .NET Framework 4.7 and 4.8. A
+        // race condition between threads reading the same PDB file to obtain file and line info for a stack trace
+        // results in AccessViolationException when the stack trace is accessed even indirectly e.g. Exception.ToString.
+        [System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptions]
+#endif
+        public static string GetMessageWithoutThrowing(this Exception exception)
+        {
+            if (exception is null) throw new ArgumentNullException(nameof(exception));
+
+            try
+            {
+                return exception.Message;
+            }
+            catch (Exception ex)
+            {
+                return ex.GetType().Name + " was thrown by the Exception.Message property.";
+            }
+        }
+
+        /// <summary>
+        /// If <see cref="Exception.Data"/> throws, returns "SomeException was thrown by the Exception.Data property."
+        /// </summary>
+#if !(NET35 || NETSTANDARD1_4)
+        // https://github.com/dotnet/coreclr/issues/19698 is also currently present in .NET Framework 4.7 and 4.8. A
+        // race condition between threads reading the same PDB file to obtain file and line info for a stack trace
+        // results in AccessViolationException when the stack trace is accessed even indirectly e.g. Exception.ToString.
+        [System.Runtime.ExceptionServices.HandleProcessCorruptedStateExceptions]
+#endif
+        public static Result<IDictionary> GetDataWithoutThrowing(this Exception exception)
+        {
+            if (exception is null) throw new ArgumentNullException(nameof(exception));
+
+            try
+            {
+                return Result.Success(exception.Data);
+            }
+            catch (Exception ex)
+            {
+                return ex.GetType().Name + " was thrown by the Exception.Data property.";
```

Comment:
Yeah, not my favorite. But I do find the explicitness of the `Result.Error` part to be desirable. I could make the syntax `Result.Error(...)` work with a new error type that implicitly converts to the generic type... 🤔

Reply:
@mikkelbu What do you think of `Result.ErrorResult`, better or worse? Should we just go with `Result.Error<IDictionary>(`?

Reply:
@jnm2 I don't mind `Result.Error<IDictionary>(...)`, but I'm not sure I follow you with respect to `Result.ErrorResult`. Was that to make `Result.ErrorResult` do "work with a new error type that implicitly converts to the generic type", as you mention above?

Reply:
I added the nested type `Result.ErrorResult` so that it is possible to write `return Result.Error(...)` instead of `return Result.Error<IDictionary>(...)` but I'm not sure whether it is worth the fanciness of making the Result API more complex. Do you have a preference?

Reply:
Sorry, I missed that you had added a commit. Since we are only using `Result.Error(...)` in one location then I prefer the more verbose version where we are explicit about the type. If we had a lot of usage then I think that the "fanciness" would be more warranted.

Reply:
Sounds sensible. Pushed a different commit, ready for review then!